docs/design_guide.md

```diff
@@ -0,0 +1,285 @@
+
+The traditional maximum line length is 79 characters for code and 72 for blocks of texts (such as docstrings, comments, etc.). These standards were created to allow for two files to fit side-by-side on a display; however, now most monitors and displays are wide enough to fit more than 160 characters across. As such, this standard will be relaxed to 100.
+
+**Blank Lines**
+
+| Code | No. of Blank Lines |
+|:---  | :---               |
+| Top level function | 2 |
+| Class Definition | 2 |
+| Methods w/ classes | 1 |
+| Logical sections w/ methods | 1-2 |
+
+
+Imports should follow the order:
+
+1. Standard library imports
+2. Related 3rd party imports
+3. Local library specific imports
+
+```python
+# Place imports on their own lines
+import os
+import sys
+
+# Submodules may be combined
+from math import floor, ceil
+
+# Absolute imports are best if possible
+# especially for imports within your own library
+import mypkg.siblin
+from mypkg import sibling2
+from mypkg.sibling import example
+```
+
+**White Space**
+
+- DO: Place 1 space immediately after commas, semicolons, and colons
+  - *Note*: Avoid placing spaces around colons which act like a binary operator (e.g. `arr[1:9]`)
+- DO NOT: Place spaces after an opening brace or before a closing brace.
+- DO NOT: Place a space before the opening parenthesis of a function call or the opening bracket for indexing/slicing
+- DO: Place 1 space on each side of arithmetic and assignment operators
+- DO NOT: Place more than 1 space before or after assignments operators, i.e. do not align `=` vertically when assigning variable values
+- DO NOT: Place spaces around the `=` sign when using with keyword arguments to a function
+- DO NOT: Leave trailing white spaces at the end of lines or in blank lines
+
+
+**Quotes**
+
+Use double quotes: `"your string"` for strings, `"""comment"""` for comments.
+
+
+### Documentation
+
+*Note: this only covers how documentation within code should be formatted, for information on how documentation for the library will be developed see [below](#library-documentation).*
+
+**Comments** should be complete sentences used to elaborate on the code.
+
+- Block comments (1 or more lines each beginning with a `# `) should be indented to match the code that *follows* them.
+- Inline Comments (a comment on the same line as the code) should be at least 2 spaces away from the statement and server a purpose. They should disambiguate the line of code, either in function or intended purpose.
+- For comments aimed at developers try to use `TODO`, `FIXME`, `NOTE`, etc. as much as possible.
```

Comment:
I'd say
> Use `TODO` and `FIXME` in comments to document future work.

TODO and FIXME are supported by tools like PyCharm which tracks their use, helps you find them etc.

I'm not familiar with any practices or support around use of the word NOTE, so I'd leave that out.

Reply:
Many editors (including Vim and Atom) support usage of other comments with `NOTE`, `QUESTION`, and more rarely some other keywords, but since it is not standard we can leave it out.